### PR TITLE
Add stakater-validate-environment 1.0.0

### PR DESCRIPTION
**Pull request by mustafaStakater**

### CODE REVIEW

**Comment by stakater-user** [comment]
@mustafaStakater Validation Successful

File: clustertask/validate-environment/templates/clustertask.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      description: Time in seconds to wait for environment
+      default: ""
+    - name: waitTimeForEnvironment
+      description: Time in seconds to wait for environment
+      default: "120"
+    - name: retryInterval
+      description: Time in seconds to get environment and check statuses
+      default: "5"
+  steps:
+  - name: push-main-tag
```
Comment (rasheedamir) [inline clustertask/validate-environment/templates/clustertask.yaml:22]: @TehreemNisa wrong name?

File: clustertask/validate-environment/templates/clustertask.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+metadata:
+  name: {{ include "stakater-validate-environment.name" . }}
+  annotations:
+    description: |
+      Validate Helm Release Status of Tronodor Environment
+spec:
+  workspaces:
+  - name: source
+  params:
+    - name: environmentName
```
[the remark this answers is not in the document]
Comment (rasheedamir) [reply clustertask/validate-environment/templates/clustertask.yaml:12]: ENVIRONMENT_NAME

File: clustertask/validate-environment/templates/clustertask.yaml [hunk 3]
```diff
@@ -0,0 +1,79 @@
+apiVersion: tekton.dev/v1beta1
+kind: ClusterTask
+metadata:
+  name: {{ include "stakater-validate-environment.name" . }}
+  annotations:
+    description: |
+      Validate Helm Release Status of Tronodor Environment
```
Comment (rasheedamir) [inline clustertask/validate-environment/templates/clustertask.yaml:7]: @mustafaStakater should we use the word HelmRelease here? Wouldn't it be better we just use word Environment

**Comment by rasheedamir** [comment]
@mustafaStakater did you manage to run and test it locally?

File: clustertask/validate-environment/templates/clustertask.yaml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        echo "Checking Environment $environmentName"
+
+        # Wait Until Timeout
+        for (( t=1; t < $(params.waitTimeForEnvironment); t++ ))
+        do
+          # Get Environment Status
+          environmentStatus=$(oc get environment $environmentName -o jsonpath='{.status}')
+
+          # Get Fields to check Environment Reconcilation
+          condition_type=$(echo $environmentStatus | jq -r '.resourceStatus.conditions[0].type')
```
Comment (rasheedamir) [inline clustertask/validate-environment/templates/clustertask.yaml:45]: @mustafaStakater why different naming convention here? Why adding _ here?

**Comment by rasheedamir** [comment]
@mustafaStakater can you plz also add the test data? and explain the readme how to run and validate it locally?

**Review by rasheedamir — CHANGES_REQUESTED** [review]
Check the feedback

**Comment by mustafaStakater** [comment]
@rasheedamir I wasnt able to test it out locally, I tried deploying the environment crd but for some reason it was continously discarding the status field because it is managed by openshift itself, i then moved on to deploying tronador but the initial setup was taking a lot of time, 
~I'll test it out on devtest for now~

File: clustertask/validate-environment/templates/clustertask.yaml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    description: |
+      Validate HelmRelease Status of Tronodor Environment
+spec:
+  workspaces:
+  - name: source
+  params:
+    - name: ENVIRONMENT_NAME
+      description: Time in seconds to wait for environment
+      default: ""
+    - name: WAIT_TIME_FOR_RECONCILE
```
Comment (rasheedamir) [inline clustertask/validate-environment/templates/clustertask.yaml:15]: TIMEOUT

**Comment by rasheedamir** [comment]
> @rasheedamir I wasnt able to test it out locally, I tried deploying the environment crd but for some reason it was continously discarding the status field because it is managed by openshift itself, i then moved on to deploying tronador but the initial setup was taking a lot of time, I'll test it out on devtest for now

as discussed this needs to be tested locally and then tests should be pushed as well in this repo

**Review by AsfaMumtaz — APPROVED** [review]

**Review by rasheedamir — APPROVED** [review]

**Pull request merged**